Assets/Scripts/Gameplay/Mecha/MechaController.cs

```diff
@@ -360,14 +429,16 @@ private void OnZoomOut(object data)
             Invoke(nameof(ResetZoomCd), juggernautParameters.scrollSensitivity / 100);
 
         }
-        
+
         private void OnRun(object data)
         {
             if (data is not bool run)
                 return;
-            _isRunning = run;
+            
+            CurrentMovementMode = run ? MovementMode.Running : MovementMode.Walking;
```

Comment:
> You will face an issue later potentially.
If you dash and move right after, you will swap from Dashing to Walking in one frame,
which may not be what you expect

Same here but for running